Refactor app menu items

pythongosssss opened the pull request:
Restructures the application menu

<img width="596" height="580" alt="image" src="https://github.com/user-attachments/assets/e1e80939-9f56-447e-a6f8-0b510cbc3c87" />

- rename Workflow to File
- move new & template items to top level
- add View menu and related sub items

Commands
- add "active" state getter shown as checkmark in the menu

Node side panel
- add refresh node defs
- change reset view icon
<img width="578" height="152" alt="image" src="https://github.com/user-attachments/assets/c5b63a82-73fa-4eef-84bb-4448beef49a8" />


Help center
- change to use store for visibility

Fixes
- Fix bug with mouse down where if you drag mouse out, mouse up wasn't caught
- Fix issue with canvas info setting not triggering a redraw on change

┆Issue is synchronized with this [Notion page](https://www.notion.so/PR-4665-Refactor-app-menu-items-2446d73d365081fda956e8b211160611) by [Unito](https://www.unito.io)

pythongosssss committed:
Restructures the application menu
- rename Workflow to File
- move new & template items to top level
- add View menu and related sub items

Commands
- add "active" state getter shown as checkmark in the menu

Node side panel
- add refresh node defs
- change reset view icon

Help center
- change to use store for visibility

Fixes
- Fix bug with mouse down where if you drag mouse out, mouse up wasn't caught
- Fix issue with canvas info setting not triggering a redraw on change

pythongosssss committed:
Fix missing translation warnings

pythongosssss committed:
Add separator under new

pythongosssss committed:
tidy

Commit:
Update locales [skip ci]

JorgeR81 commented:
In the `View` menu, I would:

 - **Remove:** `Queue Panel`, `Node Library`, `Model Library`, `Workflows`  ( unless you're planning to have and option to hide the Sidebar, when it's minimized )
 - **Remove:** `Manager`   ( unless this is supposed to show the legacy Manager )
 - **Remove:**  `Help Center` ( since this could be added to the `Help` sub menu ) 
 - **Add:** `Minimap`, `Toggle Link Visibility` ( so that we can hide the graph canvas menu, but still have these options accessible )
 
I would also add a new icon for `Toggle Link Visibility`, since it's the same as `Focus Mode`.

pythongosssss committed:
fix some tests

pythongosssss committed:
fix

pythongosssss committed:
Hide icon if there is an active state within the menu item group

pythongosssss committed:
Merge remote-tracking branch 'origin/main' into pysssss/menu-item-refactor

Commit:
Update locales [skip ci]

pythongosssss committed:
Fix tests

pythongosssss committed:
Merge branch 'pysssss/menu-item-refactor' of https://github.com/Comfy-Org/ComfyUI_frontend into pysssss/menu-item-refactor

pythongosssss committed:
Merge remote-tracking branch 'origin/main' into pysssss/menu-item-refactor

pythongosssss committed:
Implement feedback
- Remove queue, node lib, model lib, workflows, manager, help center
- Add minimap, link visibility

Commit:
Update locales [skip ci]

pythongosssss committed:
Add plus icon on "New" menu item

pythongosssss committed:
Merge branch 'pysssss/menu-item-refactor' of https://github.com/Comfy-Org/ComfyUI_frontend into pysssss/menu-item-refactor

pythongosssss committed:
Merge remote-tracking branch 'origin/main' into pysssss/menu-item-refactor

Commit:
Update locales [skip ci]

pythongosssss committed:
Fix test

pythongosssss committed:
Merge branch 'pysssss/menu-item-refactor' of https://github.com/Comfy-Org/ComfyUI_frontend into pysssss/menu-item-refactor

pythongosssss committed:
Fix translations

Commit:
Update locales [skip ci]

christian-byrne commented:
Minor conflict on the settings config and imports, sorry about that.

pythongosssss committed:
Merge remote-tracking branch 'origin/main' into pysssss/menu-item-refactor

Commit:
Update locales [skip ci]

christian-byrne reviewed (approved):
LGTM, this is great.
